modules/build/src/test/scala/scala/build/tests/BuildOptionsTests.scala

```diff
@@ -19,6 +25,167 @@ class BuildOptionsTests extends munit.FunSuite {
     )
   }
 
+  test("-S 3.nightly option works") {
+    val options = BuildOptions(
+      scalaOptions = ScalaOptions(
+        scalaVersion = Some("3.nightly"),
+        scalaBinaryVersion = None,
+        supportedScalaVersionsUrl = None
+      )
+    )
+    val scalaParams = options.scalaParams.orThrow
+    assert(
+      scalaParams.scalaVersion.startsWith("3") && scalaParams.scalaVersion.endsWith("-NIGHTLY"),
+      "-S 3.nightly argument does not lead to scala3 nightly build option"
+    )
+  }
+
+  test(s"Scala 3.${Int.MaxValue} shows Invalid Binary Scala Version Error") {
+
+    val options = BuildOptions(
+      scalaOptions = ScalaOptions(
+        scalaVersion = Some(s"3.${Int.MaxValue}"),
+        scalaBinaryVersion = None,
+        supportedScalaVersionsUrl = None
+      )
+    )
+    assert(
+      options.projectParams.isLeft && options.projectParams.left.get.isInstanceOf[
+        InvalidBinaryScalaVersionError
+      ],
+      s"specifying the 3.${Int.MaxValue} scala version does not lead to the Invalid Binary Scala Version Error"
+    )
+  }
+
+  test("Scala 2.11.2 shows Unupported Scala Version Error") {
+
+    val options = BuildOptions(
+      scalaOptions = ScalaOptions(
+        scalaVersion = Some("2.11.2"),
+        scalaBinaryVersion = None,
+        supportedScalaVersionsUrl = None
+      )
+    )
+    assert(
+      options.projectParams.isLeft && options.projectParams.left.get.isInstanceOf[
+        UnsupportedScalaVersionError
+      ],
```

Comment:
Better avoid `Either.get` and `asInstanceOf`. Same for all their uses below.

```suggestion
      options.projectParams.swap.exists { case _: UnsupportedScalaVersionError => true; case _ => false },
```